### PR TITLE
Serach engine functionality

### PR DESCRIPTION
**Pull request by HatemMorgan**

### CODE REVIEW

File: src/main/java/com/linkedin/replica/serachEngine/databaseHandlers/impl/ArangoHandler.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+		name = name.toLowerCase();
+		
+		// query for getting user if there is a match of his/her firstName or lastName with input
+		String query = "FOR t IN "+ collectionName +" FILTER "
+				+ "LOWER(t.firstName) LIKE @name || "
+				+ "LOWER(t.lastName) LIKE  @name "
+				+ "RETURN t";
+		
+		// bind variables in query
+		Map<String, Object> bindVars = new MapBuilder().put("name", "%"+name+"%").get();
```
Comment (mostafa-abdullah) [inline src/main/java/com/linkedin/replica/serachEngine/databaseHandlers/impl/ArangoHandler.java:55]: Can you please adjust the indentation in this file? Make sure that the whitespaces consist of only tabs not spaces

File: src/main/java/com/linkedin/replica/serachEngine/databaseHandlers/impl/SqlHandler.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import com.linkedin.replica.serachEngine.databaseHandlers.DatabaseHandler;
+import com.linkedin.replica.serachEngine.models.Company;
+import com.linkedin.replica.serachEngine.models.Job;
+import com.linkedin.replica.serachEngine.models.User;
+import com.linkedin.replica.serachEngine.models.Post;
+
+/**
+ * Implementation of SqlHandler which is responsible for serving specific request from SQL database
+ */
+public class SqlHandler implements DatabaseHandler {
```
Comment (mostafa-abdullah) [inline src/main/java/com/linkedin/replica/serachEngine/databaseHandlers/impl/SqlHandler.java]: Do we need this file? I think SQL is not concerned with any search functionality

File: src/main/java/com/linkedin/replica/serachEngine/models/Post.java [hunk 3]
```diff
@@ -4,5 +4,33 @@
  * Post Model
  */
 public class Post {
+	private String id;
+	private String text;
+	private String authorId;
```
Comment (mostafa-abdullah) [inline src/main/java/com/linkedin/replica/serachEngine/models/Post.java:9]: Why do we have the IDs as strings? didn't we agree that they will be `int` to be compatible with corresponding entries in SQL?

**Review by mostafa-abdullah — COMMENTED** [review]
Only very minor comments. VERY great work. 3assh gddan wallahi :muscle: :smile:

File: src/main/java/com/linkedin/replica/serachEngine/databaseHandlers/impl/SqlHandler.java (hunk 2 above)
Comment (HatemMorgan) [reply src/main/java/com/linkedin/replica/serachEngine/databaseHandlers/impl/SqlHandler.java]: I kept it as to be as a reference for everyone else

**Pull request merged**